### PR TITLE
Enforce reader permissions

### PR DESCRIPTION
**Pull request by midsum-salrux**
#1195 

This doesn't prevent seeing that secret channels exist yet, but you can't join them.

### CODE REVIEW

File: desk/app/groups.hoon [hunk 1]
```diff
@@ -635,13 +637,15 @@
       ?+    rest.pole  ~
           [%can-read src=@ ~]
         =/  src  (slav %p src.rest.pole)
-        ?:  =(%open -.cordon.group)  `loob+!>(&)
```
Comment (midsum-salrux) [inline desk/app/groups.hoon:638]: Note the removal of this check. Before, you could read everything in an open group, even secret channels.

Comment (arthyn) [reply desk/app/groups.hoon:638]: good catch

**Review by arthyn — APPROVED** [review]
this looks good to me as a first step. 

I'm wondering if it's possible to even prevent the channels themselves from ending up on the ship without permissions. right now we give all group updates unequivocally, but it seems like if a channel add comes in that's admin only it shouldn't even go to the receiving ship that's lacking. however, this makes sending updates much more complicated. I guess we could just guard for it on the receiving ship and if someone modifies their source to see them that's acceptable. @midsum-salrux @ryjm @liam-fitzgerald what do y'all think?

**Comment by midsum-salrux** [comment]
> I'm wondering if it's possible to even prevent the channels themselves from ending up on the ship without permissions. right now we give all group updates unequivocally, but it seems like if a channel add comes in that's admin only it shouldn't even go to the receiving ship that's lacking. however, this makes sending updates much more complicated. I guess we could just guard for it on the receiving ship and if someone modifies their source to see them that's acceptable. @midsum-salrux @ryjm @liam-fitzgerald what do y'all think?

yeah, this is what I ran into. I'm OK with the solution you mentioned since even if they did modify their local source, they could only see the name of the secret channel, not actually join it.

**Review by ryjm — APPROVED** [review]
LGTG

 (sidenote: it's relatively easy to create channels associated with a group without initializing them. they'll never appear in the UI but show up in a `:chat +dbug` for example. might want to account for that at some point too)

**Pull request merged**